### ISSUE TITLE
Accessibility Modal cannot be dismissed by re-clicking on the Accessibility icon

### ISSUE THREAD

**Issue by ryanfchase** [hackforla/311-data#1809]
### Overview

We need the accessibility modal button to open and close on-click so that users can dismiss it predictably.

### Action Items

- [x] (code health*) refactor `isIconOpen`
  - [x] dev opted to use a single variable, this is OK ~add a new variable to the Header function component, e.g. `iconButtonRef` via `React.useRef`. Use this variable to track the DOM element of the IconButton~
  - [x] same comment as above ~modify `isIconOpen` to only take on values: `true` or `false`~
  - [x] optional: consider renaming `isIconOpen` to `isMenuOpen` (since it is the menu that opens, rather than the icon)
- [x] modify the `onClick` behavior of IconButton to account for whether the Menu is open or closed
- [x] Upload SCREEN RECORDING of accessibility modal working correctly

Note on code health*, in Header.jsx, `isIconOpen` currently takes on 2 roles:
- whether or not the menu is currently open
  - L160: `aria-controls={isIconOpen ? 'accessibile-menu' : undefined}`
  - L162: `aria-expanded={isIconOpen ? 'true' : undefined}`
  - L173: `open={Boolean(isIconOpen)}`
- it represents the DOM element that the Menu should attach to
  - L132: `setIsIconOpen(event.currentTarget)`
  - L136:  `setIsiconOpen(null)`
  - L170: `anchorEl={isIconOpen}`

##### External Docs
- [React: useRef](https://react.dev/reference/react/useRef)
- [Material UI: Menu API](https://mui.com/material-ui/api/menu/)
- [Stack Overflow: using a ref with Menu component](https://stackoverflow.com/questions/48157863/how-to-make-a-dropdown-menu-open-below-the-appbar-using-material-ui)

##### Useful Files and Components
Header: `components/Header.jsx`
  - see IconButton, L157: `<IconButton ... />`
    - review the `onClick` prop
  - see Menu, L167: `<Menu ... />`
  - review `handleIconClick` and `handleIconClose`

<details><summary>SCREEN RECORDING after implementing changes</summary>
<p>

https://github.com/user-attachments/assets/0a4cc32d-fea7-4d1c-8b18-4d89a8024f2d

</p>
</details>

**Comment by Brandoncyu** [comment]
https://github.com/user-attachments/assets/0a4cc32d-fea7-4d1c-8b18-4d89a8024f2d

**Comment by ryanfchase** [comment]
@Brandoncyu thank you for working on this! I've provided a review, please have a look when you get a chance. I've also made an edit to your PR, I can cover that and other ticket-formatting notes at the meeting tmo night 👍
